### ISSUE TITLE
Incorrect data out of bounds check

### ISSUE THREAD

**Issue by Klaama** [miscavage/object-to-csv#1]
Thanks for providing this library!

When I was using this library, I found the data out of bounds check in _getDataForKeyByIndex(index, key) was not right:
```
if (this._keys.length < index - 1) {
    return undefined;
}
```
Probably, '`this._data.length`' should be used instead? Otherwise, I had only several data entries got converted to CSV string, other entries were all "undefined,undefined,..."

**Comment by ivan-fuentes** [comment]
Same issue here, changing the comparison as @Klaama indicates corrects the issue.

**Comment by miscavage** [comment]
Thank for you flagging @Klaama, @ivan-fuentes. This has been resolved with PR#2 and is published to master @ v1.0.3.
